pages/api/globalSearch.ts

```diff
@@ -27,58 +28,73 @@ export const config = {
-  const headers = {
-    headers: {
-      'x-api-key': apiKey || '',
-    },
-  }
+  let queryParams: paths['/search/collections/v1']['get']['parameters']['query'] =
+    {
+      name: q as string,
+      limit: 6,
+    }
 
-  let queryParams: paths['/search/collections/v1']['get']['parameters']['query'] = {
-    name: q as string,
-    limit: 6
+  if (COLLECTION_SET_ID) {
+    queryParams.collectionsSetId = COLLECTION_SET_ID
+  } else if (COMMUNITY) {
+    queryParams.community = COMMUNITY
   }
 
-  if (COLLECTION_SET_ID) {
-   queryParams.collectionsSetId = COLLECTION_SET_ID
- } else if (COMMUNITY){
-   queryParams.community = COMMUNITY
- }
-
-  // start fetching search preemptively
-  let collectionQuery = fetcher(
-    `${reservoirBaseUrl}/search/collections/v1`,
-    queryParams,
-    headers
-  )
+  supportedChains.forEach(async (chain) => {
+    const { reservoirBaseUrl, apiKey } = chain
+    const headers = {
+      headers: {
+        'x-api-key': apiKey || '',
+      },
+    }
+    promises.push(
+      fetcher(`${reservoirBaseUrl}/search/collections/v1`, queryParams, headers)
+    )
+  })
+
+  let isAddress = ethers.utils.isAddress(q as string)
 
   if (isAddress) {
-    let { data } = await fetcher(
-      `${reservoirBaseUrl}/collections/v5?contract=${q}&limit=6`,
-      {},
-      headers
-    )
-    if (data.collections.length) {
-      searchResults = data.collections.map((collection: Collection) => {
-        let processedCollection: SearchCollection = {
+    const promises = supportedChains.map(async (chain) => {
+      const { reservoirBaseUrl, apiKey } = chain
+      const headers = {
+        headers: {
+          'x-api-key': apiKey || '',
+        },
+      }
+      const { data } = await fetcher(
+        `${reservoirBaseUrl}/collections/v5?contract=${q}&limit=6`,
+        {},
+        headers
+      )
+      return data.collections.map((collection: Collection) => {
+        const processedCollection: SearchCollection = {
           collectionId: collection.id,
           contract: collection.primaryContract,
           image: collection.image,
           name: collection.name,
           allTimeVolume: collection.volume?.allTime,
           floorAskPrice: collection.floorAsk?.price?.amount?.decimal,
-          openseaVerificationStatus: collection.openseaVerificationStatus
+          openseaVerificationStatus: collection.openseaVerificationStatus,
+          chainName: chain.name.toLowerCase(),
+          chainId: chain.id,
+          searchIcon: chain.searchIcon,
         }
         return {
-        type: 'collection',
-        data: processedCollection,
-      }})
+          type: 'collection',
+          data: processedCollection,
+        }
+      })
+    })
+
+    let results = await Promise.all(promises).then((results) => results.flat())
+
+    if (results.length > 0) {
+      searchResults = results
     } else {
       let ensData = await fetch(
```

Comment:
Maybe we should also call this async and always return it in the data. Curious to get Lingzi's idea on this.

Reply:
That's a good thought. I will ask Lingzi what she thinks